clients/java/src/main/java/io/camunda/zeebe/client/impl/command/AddPermissionsCommandImpl.java

```diff
@@ -41,49 +39,42 @@ public class AddPermissionsCommandImpl
         AddPermissionsCommandStep3,
         AddPermissionsCommandStep4 {
 
-  private final String path;
-  private final AuthorizationPatchRequest request;
-  private final JsonMapper jsonMapper;
+  private final PatchAuthorizationCommand delegate;
   private final HttpClient httpClient;
+  private final JsonMapper jsonMapper;
+  private final String path;
   private final RequestConfig.Builder httpRequestConfig;
-  private PermissionDTO currentPermission;
 
   public AddPermissionsCommandImpl(
       final long ownerKey, final HttpClient httpClient, final JsonMapper jsonMapper) {
     this.httpClient = httpClient;
     this.jsonMapper = jsonMapper;
-    httpRequestConfig = httpClient.newRequestConfig();
-    request = new AuthorizationPatchRequest().action(ActionEnum.ADD);
     path = "/authorizations/" + ownerKey;
+    httpRequestConfig = httpClient.newRequestConfig();
+    delegate = new PatchAuthorizationCommand(ActionEnum.ADD);
```

Comment:
❌ The default request timeout should be set in the constructor as well. Please have a look at other commands, like `SetVariables` or `DeployResources`

Reply:
@nicpuppa Are you sure this is required? From what I can see I have a request config here which is created on line 53:

```java
httpRequestConfig = httpClient.newRequestConfig();
```

If we look in the implementation of this method:

```java
   * Creates a new request configuration builder with the default values. The builder can be used to
   * customize the request configuration for a specific request.
   *
   * @return a new request configuration builder
   */
  public RequestConfig.Builder newRequestConfig() {
    return RequestConfig.copy(defaultRequestConfig);
  }
```

This default request config is built like:

```java
  private Builder defaultClientRequestConfigBuilder() {
    return RequestConfig.custom()
        .setResponseTimeout(Timeout.of(config.getDefaultRequestTimeout()))
        // TODO: determine if the existing (gRPC) property makes sense for the HTTP client
        .setConnectionKeepAlive(TimeValue.of(config.getKeepAlive()))
        // hard cancellation may cause other requests to fail as it will kill the connection; can be
        // enabled when using HTTP/2
        .setHardCancellationEnabled(false);
  }
```

So the default request timeout is already set on this, no?

Reply:
There's a bunch of other commands where we also don' set this. Correlate message and Clock pin for example

Reply:
The config is passed along to the request. It's already on the request.

Reply:
Sorry, I didn't saw your comment above. I wasn't aware that we are setting the default while building the config (but it makes total sense). All good then, we should probably get rid of the duplicates in each commands implementation

Reply:
>I wasn't aware that we are setting the default while building the config

Definitely don't git blame it 😉

Reply:
Yeah saw that, but git blame is laying. All the files in the java client were moved when we wanted to replace the Zeebe Client with the Camunda Client, this is why almost all the files in the java client are under my name 😄